src/WebJobs.Script.WebHost/TokenExpiration/TokenExpirationService.cs

```diff
@@ -0,0 +1,184 @@
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                _cancellationTokenSource?.Cancel();
+
+                if (_analysisTask != null && !_analysisTask.IsCompleted)
+                {
+                    _logger.LogDebug("Token expiration service stopped before analysis completion. Waiting for cancellation.");
+
+                    return _analysisTask;
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error stopping Token expiration service. Handling error and continuing.");
+            }
+            return Task.CompletedTask;
+        }
+
+        private void ScheduleTokenExpirationCheck()
+        {
+            _analysisScheduled = true;
+            _cancellationTokenSource = new CancellationTokenSource();
+
+            _analysisTask = Task.Delay(TimeSpan.FromSeconds(5), _cancellationTokenSource.Token)
+               .ContinueWith(t => AnalyzeSasTokenInUri());
+        }
+
+        private void AnalyzeSasTokenInUri()
+        {
+            var jobHost = _scriptHost.GetService<IScriptJobHost>();
+
+            if (jobHost == null
+                || _cancellationTokenSource.IsCancellationRequested)
+            {
+                return;
+            }
+
+            // WEBSITE_RUN_FROM_PACKAGE and WEBSITE_CONTENTAZUREFILECONNECTIONSTRING
+            Dictionary<string, string> appSettings = new Dictionary<string, string>
+            {
+                { EnvironmentSettingNames.AzureWebsiteRunFromPackage, _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteRunFromPackage) },
+                { EnvironmentSettingNames.AzureFilesConnectionString, _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureFilesConnectionString) }
+            };
+
+            // We only want to parse AzureWebJobsStorage in the following format
+            // BlobEndpoint=<blob-endpoint-url>;QueueEndpoint=<queue-endpoint-url>;TableEndpoint=<table-endpoint-url>;SharedAccessSignature=<sas-token>
+            var azureWebJobsStorage = _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebJobsSecretStorage);
+
+            // Check if the value starts with "BlobEndpoint=" and contains "SharedAccessSignature="
+            if (azureWebJobsStorage.StartsWith("BlobEndpoint=", StringComparison.OrdinalIgnoreCase)
+                && azureWebJobsStorage.Contains("SharedAccessSignature="))
+            {
+                appSettings.Add(EnvironmentSettingNames.AzureWebJobsSecretStorage, azureWebJobsStorage);
+            }
+
+            EmitLogging(appSettings);
+        }
+
+        private void EmitLogging(Dictionary<string, string> appSettings)
+        {
+            var currentDate = DateTime.Now.Date;
+            foreach (var setting in appSettings)
+            {
+                var uri = setting.Value;
+                if (!string.IsNullOrEmpty(uri))
```

Comment:
We check the app setting value to ensure that it is not null or empty. Given that this app setting can be set by the user, should we check for white space (`string.IsNullOrWhiteSpace(...)`) as well?